include/drivers/dma.h

```diff
@@ -77,9 +77,14 @@ enum dma_addr_adj {
  *     reserved           [ 13 : 15 ]
  */
 struct dma_block_config {
+#ifdef CONFIG_DMA_64BIT
+	uint64_t source_address;
+	uint64_t dest_address;
+#else
 	uint32_t source_address;
-	uint32_t source_gather_interval;
 	uint32_t dest_address;
+#endif
+	uint32_t source_gather_interval;
 	uint32_t dest_scatter_interval;
```

Comment:
I don't know much about DMA, but in case of 64 bit src/dst: could these intervals go beyond 32 bits?

Reply:
I am not aware of usage of interval fields  and in the current pl330 driver only one src/dst address pair is supported.
So didn't touch those fields. No other DMA driver's are using these fields as of now.